### PR TITLE
Fix crash when calling `get_video_adapter_*` in a thread

### LINKED ISSUE #79503: Game crash when using compatibility rendering method
### Godot version

4.1 and 4.0.3 debug build

### System information

OS: Manjaro Linux x86_64
Kernel: 6.1.38-1-MANJARO 
CPU: AMD Ryzen 5 5600X (12) @ 3.700GHz 
GPU: NVIDIA GeForce RTX 3070 Ti
X11 backend
OpenGL API 3.3.0 NVIDIA 535.54.03

### Issue description

Game crashes silently after switching to compatibility rendering method. 
It's a pure 2D game. Forward+ and mobile work fine.

logs in console:

[4.0.3 debug build](https://github.com/Calinou/godot-debug-builds):
```
================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.0.3.stable.custom_build (5222a99f5d38cd5346254cefed8f65315bca4fcb)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] /usr/lib/libc.so.6(+0x39ab0) [0x7fe425c9dab0] (??:0)
[2] /usr/lib/libGLdispatch.so.0(+0x4d6c9) [0x7fe4176de6c9] (??:0)
-- END OF BACKTRACE --
================================================================

```

4.1 official:
```
================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.1.stable.official (970459615f6b2b4151742ec6d7ef8559f87fd5c5)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] /usr/lib/libc.so.6(+0x39ab0) [0x7fc48dfd9ab0] (??:0)
[2] /usr/lib/libGLdispatch.so.0(+0x4d6c9) [0x7fc48462b6c9] (??:0)
-- END OF BACKTRACE --
================================================================

```

### Steps to reproduce

Switch to compatibility rendering method and run game in editor.

### Minimal reproduction project
[signal11.zip](https://github.com/godotengine/godot/files/12061947/signal11.zip)

### PR DESCRIPTION
**Pull request by univeous**
Fixes #79503 .

### CODE REVIEW

**Review by RandomShaper — APPROVED** [review]

File: servers/rendering/rendering_server_default.h [hunk 1]
```diff
@@ -945,10 +945,27 @@ class RenderingServerDefault : public RenderingServer {
 	FUNC0(global_shader_parameters_clear)
 
 #undef server_name
+#undef ServerName
+	/* STATUS INFORMATION */
+#define ServerName RendererUtilities
+#define server_name RSG::utilities
+	FUNC0RC(String, get_video_adapter_name)
+	FUNC0RC(String, get_video_adapter_vendor)
+	FUNC0RC(RenderingDevice::DeviceType, get_video_adapter_type)
```
Comment (clayjohn) [inline servers/rendering/rendering_server_default.h:954]: This shouldn't be included here. ``FUNC0RC`` forces functions to be called on the command queue which adds unnecessary performance overhead to functions if they don't need to be called from the rendering thread.

``get_video_adapter_type`` should stay below with the other status information functions

**Comment by univeous** [comment]
Oh sorry I messed up, will fix it soon🥲

**Review by clayjohn — APPROVED** [review]
Looks good to me!

**Comment by YuriSizov** [comment]
Thanks!

**Comment by YuriSizov** [comment]
Cherry-picked for 4.1.2.

**Pull request merged**